src/lib/asn1/ASN1Reader.cpp

```diff
@@ -285,32 +254,16 @@ CHIP_ERROR ASN1Reader::GetBitString(uint32_t & outVal)
 CHIP_ERROR ASN1Reader::DecodeHead()
 {
     const uint8_t * p = mElemStart;
-
-    if (p >= mBufEnd)
-        return ASN1_ERROR_UNDERRUN;
+    ReturnErrorCodeIf(p + 1 >= mBufEnd, ASN1_ERROR_UNDERRUN);
```

Comment:
Could we be in a state where `p` == `mBufEnd` but `p+1` overflows?  The old code handled this; the new code seems like it would not.